packages/gatsby/src/utils/worker/pool.ts

```diff
@@ -91,6 +92,10 @@ export async function runQueriesInWorkersQueue(
         .catch(handleRunQueriesInWorkersQueueError)
     }
 
+    // at this point, we're done grabbing page dependencies, so we need to
+    // flush out the batcher in case there are any left in the queue
+    createPageDependencyBatcher.flush()
```

Comment:
I'd need to jog my memory, but I think we need to use `.flush()` inside each worker process (maybe as part of `saveQueriesDependencies` function we use below) and not on "main" process. In PQR mode `CREATE_COMPONENT_DEPENDENCY` actions are happening inside workers that run queries, not on main one.

Reply:
Ahhh right right, good call!